packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -185,8 +185,12 @@ class TextPainter {
        _textWidthBasis = textWidthBasis,
        _textHeightBehavior = textHeightBehavior;
 
+  // _paragraph being null means the text needs layout because of style changes.
+  // Setting _paragraph to null invalidates all the layout cache.
   ui.Paragraph? _paragraph;
-  bool _needsLayout = true;
+  // Whether _paragraph contains outdated paint information and needs to be
+  // rebuilt.
+  bool _needsPaint = true;
```

Comment:
It's a little confusing that the property is called needsPaint, but the comment talks about needing rebuild...